### DISCUSSION TITLE
Different behaviors between two GET request methods in Django Ninja TestClient

### DISCUSSION THREAD

**Discussion by virgil724** [vitalik/django-ninja#1369] [Q&A]
## Issue Description

I've encountered different test results when using these two GET request approaches with Django Ninja TestClient:
```python
self.client = TestClient(router)

# Method 1: Using URL query string
response = self.client.get(f"/draw_winner?prize={self.prize.id}")

# Method 2: Using data parameter
response = self.client.get("/draw_winner", data={"prize": self.prize.id})
```
While these two methods appear to construct the same HTTP GET request, they produce different test results.
## Questions

What are the underlying implementation differences between these two approaches?
Why do they lead to different test results?
In which scenarios should each approach be preferred?

## Environment

Django Ninja Test Client
Python version: 3.12
Django Ninja version: 1.3.0
Django version: 5.1.1
## Code Example
```python
@router.get(
    "/draw_winner", response={200: WinnerRecordSchema, 400: Message}, tags=["winner"]
)
def draw_winner(request, prize: int = None, filters: EmployeeFilterSchema = Query(...)):
  print(prize)
self.client = TestClient(router)

# Test example
response = self.client.get(f"/draw_winner?prize={self.prize.id}")
# Will Print Out self.prize.id

response = self.client.get("/draw_winner", data={"prize": self.prize.id})
# Will Print out None
```
If you've encountered similar issues or understand the differences between these approaches, please share your insights and experiences.
Would you like me to modify or add anything to make the issue clearer?

**Comment by vitalik** [comment]
Hi @virgil724 
yeah indeed there is a confusion - `data` is only used for POST request (basically populating request.POST )

I guess it's better make it better compatibility with standard django test client by adding `query_params` argument
